### PR TITLE
feat (converter/core): Enhanced JUEL converter

### LINKED ISSUE #60: Improve the JuelToFeelConverter in Backend
### Is your feature request related to a problem? Please describe.

Currently, the Juel to Feel converter is very simple and does not cover all relevant cases.

### Describe the solution you'd like

Port the existing js converter to Java

### Describe alternatives you've considered

Use the existing JS converter with the embedded JS engine

### Additional context

no

### PR DESCRIPTION
**Pull request by jonathanlukas**
closes #60 

## Description
The JUEL converter that was part of the Camunda Modeler Plugin was more advanced than the one that was embedded in Java.

I took both of them and created an immortal beast.

## Additional context
When converting diagrams, it might be required to transform expressions.

Of course, they are always subject to review, but the more we can handle, the better.

## Testing your changes
There is an extensive unit test for many different juel expressions that are transformed to become feel expressions.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an existing open issue)
- [x] New feature (non-breaking change which adds functionality to an extension)
- [ ] Breaking change (fix or feature that would cause existing functionality of an extension to change)
- [ ] Documentation update (changes made to an existing piece of documentation)

## Checklist:
- [x] My code adheres to the syntax used by this extension.
- [ ] My pull request requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **Camunda Community Hub** documentation.
- [x] I have read the **Pull Request Process** documentation.
- [x] I have added or suggested tests to cover my changes suggested in this pull request.
- [x] All new and existing CI/CD tests passed.
- [x] I will /assign myself this issue, and add the relevant [issue labels] to it if they are not automatically generated by Probot.
- [x] I will tag @camunda-community-hub/devrel in a new comment on this issue if 30 days have passed since my pull request was opened and I have not received a response from the extension's maintainer.

### CODE REVIEW

File: backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java [hunk 1]
```diff
@@ -1,47 +1,88 @@
...
-    String undeclaredExpression = getUndeclaredExpression(expression);
-    if (isTraversingExpression(expression) && !isExpression(undeclaredExpression)) {
-      return Optional.of("=" + undeclaredExpression);
+  private String handleExpression(String expression) {
+    // replace operators globally
+    String replaced =
+        expression
+            .replaceAll("not ", "!")
+            .replaceAll("\\[\"", ".")
+            .replaceAll("\"\\]", "")
```
Comment (ingorichtsmeier) [inline backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java:57]: Should be handled by regex groups

**Review by ingorichtsmeier — CHANGES_REQUESTED** [review]
improve regex handling

File: backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java [hunk 2]
```diff
@@ -1,47 +1,80 @@
...
-  public static Optional<String> transform(String possibleExpression, boolean createExpression) {
-    if (possibleExpression == null) {
-      return Optional.empty();
+  public static String transform(final String expression, DomElementVisitorContext context) {
+    String transform = INSTANCE.doTransform(expression);
+    if (context != null) {
+      context.addMessage(
+          Severity.TASK,
+          "Please review transformed expression: '" + expression + "' -> '" + transform + "'");
     }
```
Comment (ingorichtsmeier) [inline backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java:21]: This nearly doubles the amount of tasks, as most transformation include a feel expression as well:
```
- TASK: Please review transformed expression: 'management' -> 'management'
- TASK: Attribute 'candidateGroups' on 'userTask' was mapped, please review. Hint: Please review candidate groups 'management'
```
Could we merge this into a single Task?
Like 
```
- TASK: Attribute 'candidateGroups' on 'userTask' was mapped, please review. Hint: Please review candidate groups with new expression `management`
```

**Review by ingorichtsmeier — COMMENTED** [review]
Maybe we can merge it as it looks good and move my comment into a separate issue.

What do you think?

File: backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java (hunk 2 above)
Comment (jonathanlukas) [reply backend-diagram-converter/core/src/main/java/org/camunda/community/converter/ExpressionUtil.java]: This is better now.

**Pull request merged**